jquery.bemhelpers.js

```diff
@@ -62,40 +62,50 @@
                         return false;
                     }
                 });
-                
+
                 return modVal;
-                
+
             } else return undefined;
         },
-        
+
         setMod: function(block, elem, modName, modVal) {
             if (modVal === undefined) {
                 modVal = modName;
                 modName = elem;
                 elem = undefined;
             }
-            
+
             return this.each(function() {
                 var $this = $(this),
                     classPattern = block + (elem !== undefined ? BEMsyntax.elem + elem : '') +
                         BEMsyntax.modBefore + modName;
-                
+
+                var removeTheAlreadySetValue = false;
+
                 if (modVal === false) {
                     $this.removeClass(classPattern);
                 } else if (modVal === true) {
                     $this.addClass(classPattern);
                 } else {
+                    // if the String modifer value is an empty String, then it means that user likes to remove it. So set 'removeTheAlreadySetValue' to true.
+                    if (modVal === '') removeTheAlreadySetValue = true;
+
                     $.each(getElemClasses(this), function(i, c) {
                         if (c.indexOf(classPattern) === 0
                             && c.substr(classPattern.length, BEMsyntax.modKeyVal.length) === BEMsyntax.modKeyVal) {
 
                             $this.removeClass(c);
                         }
                     });
-                    
-                    $this.addClass(classPattern + BEMsyntax.modKeyVal + modVal);
+
+                    // Check whether we should remove the modifer class or not according to 'removeTheAlreadySetValue'
+                    if (removeTheAlreadySetValue === true) $this.removeClass(classPattern + BEMsyntax.modKeyVal + modVal);
+                    else $this.addClass(classPattern + BEMsyntax.modKeyVal + modVal);
                 }
-                
+
+                // if we're dealing with a String modifer(ONLY if we're dealing with a String modifer 'removeTheAlreadySetValue' may be true) and 'removeTheAlreadySetValue' is true, then it means that user aims to remove the already existing modifier value class.
+                if (removeTheAlreadySetValue === true) modVal = false;
```

Comment:
I'd say let's keep it consistent. If we make an empty string a special case (i.e. it removes the whole key-value modifier), let's also broadcast the same empty string back. This could also potentially make comparisons easier for the event listeners as we stick to the same data type.